libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/coordinator/MapLayoutCoordinator.kt

```diff
@@ -49,10 +50,10 @@ internal class MapLayoutCoordinator(
                         viewGroup.context,
                     ).enter()
 
-                    val binding = MapboxMapviewLayoutBinding.bind(viewGroup)
-                    initDefaultMap(binding.mapView.getMapboxMap())
-                    mapViewOwner.updateMapView(binding.mapView)
-                    binding.mapView
+                    val mapView = BoundMapViewProvider.bindLayoutAndGet(viewGroup)
```

Comment:
```suggestion
                    val mapView = MapboxMapviewLayoutBinding.bind(viewGroup).mapView
```
I don't think we need BoundMapViewProvider. Let's inline it.

Reply:
I see what you mean. We need to address the root cause, which is the incorrect placement of the layout inflation code. Notice that everywhere else, we inflate layouts only in Binders. Here, however, we do this in the `flow.map` transform.

I propose encapsulating all map layout inflation logic in the new `MapLayoutBinder` that can be subclassed to allow custom `MapView` injection (similar to what `InfoPanelBinder` does). Then refactor `MapLayoutCoordinator` to just: 

```kotlin
override fun MapboxNavigation.flowViewBinders(): Flow<Binder<ViewGroup>> {
    return navigationViewContext.uiBinders.mapLayoutBinder.map { mapLayoutBinder ->
        (mapLayoutBinder ?: MapLayoutBinder.defaultBinder()).also {
            // inject NavigationViewContext and MapboxNavigationViewLayoutBinding
            it.setNavigationContext(navigationViewContext)
            it.setNavigationViewBinding(binding)
        }
    }
}
```

If you are up for a challenge, go for it. Otherwise, don't worry about it. I'll do this some other time.

Reply:
I can try. Maybe in another PR?

Reply:
#6433.